web/src/routes/admin/(resources)/rooms/+page.svelte

```diff
@@ -1,25 +1,49 @@
 <script lang="ts">
-  import DataTable from '$lib/elements/ui/dataTable/data-table.svelte';
-  let columnNames = ['Name', 'Gebäude', 'Kapazität', 'test'];
+  import DataTable, { type DataItem } from '$lib/elements/ui/dataTable/data-table.svelte';
+  import { deleteRoom, getRooms, type Pageable, type PageRoom } from '$lib/sdk/fetch-client';
+  import { error } from '@sveltejs/kit';
+  import { onMount } from 'svelte';
+  import { writable, type Writable } from 'svelte/store';
 
-  //demo data from shadcn
-  const data = [
-    {
-      id: '728ed52f',
-      amount: 100,
-      status: 'pending',
-      email: 'm@example.com',
-    },
-    {
-      id: '489e1d42',
-      amount: 125,
-      status: 'processing',
-      email: ['test1', 'test2'],
-    },
-  ];
+  let columnNames = ['Name', 'Gebäude', 'Kapazität', 'Tags'];
+  let keys = ['id', 'Name', 'buildingName', 'capacity', 'tags'];
+  let tableData: Writable<DataItem[]> = writable([]);
+  let totalElements: Writable<number> = writable(0);
+
+  onMount(async () => await loadPage(0, '', ''));
+
+  async function loadPage(index: number, sortString: string, filter: string) {
+    let pageable: Pageable = { page: index, size: 10, sort: [sortString] };
+    let result: PageRoom;
+    try {
+      /^\d+$/.test(filter)
+        ? (result = await getRooms(pageable, {
+            name: filter,
+            buildingName: filter,
+            capacity: filter as unknown as number,
+          }))
+        : (result = await getRooms(pageable, {
+            name: filter,
+            buildingName: filter,
+          }));
+      totalElements.set(result.totalElements as number);
+      tableData.set(result.content as unknown as DataItem[]);
+    } catch {
+      error(404, { message: 'Could not fetch page' });
```

Comment:
Not sure if a 404 is fitting here. What were your thoughts?

Reply:
Why wouldn't 404 be fitting here? What error code would you choose instead?